core/src/main/java/org/infinispan/interceptors/distribution/TxDistributionInterceptor.java

```diff
@@ -182,7 +183,7 @@ public Object visitReadWriteKeyCommand(InvocationContext ctx, ReadWriteKeyComman
 
    @Override
    public Object visitWriteOnlyManyEntriesCommand(InvocationContext ctx, WriteOnlyManyEntriesCommand command) throws Throwable {
-      return handleTxWriteManyEntriesCommand(ctx, command, command.getEntries(), (c, entries) -> new WriteOnlyManyEntriesCommand(c).withEntries(entries));
+      return handleTxWriteManyEntriesCommand(ctx, command, command.getEntries(), (c, entries) -> new WriteOnlyManyEntriesCommand(c, componentRegistry).withEntries(entries));
```

Comment:
You've made the lambda capture `this` reference; please store the lambda in a field. Ofc if you don't remove the `componentRegistry` param completely, see my comment on ReadOnlyManyCommand